bundles/org.pitest.pitclipse.runner/src/org/pitest/pitclipse/runner/PitRunner.java

```diff
@@ -16,88 +16,37 @@
 
 package org.pitest.pitclipse.runner;
 
-import com.google.common.base.Function;
-import com.google.common.base.Optional;
-import com.google.common.collect.ImmutableList;
+import java.io.File;
 
 import org.pitest.mutationtest.commandline.MutationCoverageReport;
-import org.pitest.pitclipse.runner.client.PitClient;
 import org.pitest.pitclipse.runner.results.Mutations;
 import org.pitest.pitclipse.runner.results.mutations.RecordingMutationsDispatcher;
+import org.pitest.pitclipse.runner.util.PitFileUtils;
 
-import java.io.File;
-import java.io.IOException;
-
-import static com.google.common.base.Preconditions.checkArgument;
-import static java.lang.Integer.parseInt;
+import com.google.common.base.Function;
 
 /**
  * Executes PIT.
  */
 public class PitRunner {
 
-    public static void main(String[] args) {
-        validateArgs(args);
-        int port = parseInt(args[0]);
-        
-        try (PitClient client = new PitClient(port)) {
-            client.connect();
-            Optional<PitRequest> request = client.readRequest();
-            Optional<PitResults> results = request.transform(executePit());
-
-            results.toJavaUtil().ifPresent(client::sendResults);
-            
-        } catch (IOException e) {
-            // An error occurred while closing the client
-            e.printStackTrace();
-        }
+    private PitRunner() {
+        // Only static methods
```

Comment:
Since this class now provides only a single public method, couldn't we make it implement the `Function<PitRequest, PitResults>`? We'd use it as follows in `PitRunnerMain`:

```java
Optional<PitResults> results = request.map(PitRunner::new);
```

and I feel that's a better design as a class implementing an interface provides better extensibility than static methods.

I think I'd even renamed the class `ExecutePit` or `PitExecution` but that's just a personal preference about naming things. Feel free to keep `PitRunner` if you think it's a better name.

Reply:
Good point!
I'm doing that. However, it looks like Java doesn't like the constructor reference: you have to pass a new instance. Indeed, `PitRunner::new` corresponds to a `Provider`, while `map` actually wants a `Function`:

```java
Optional<PitResults> results = request.map(new PitRunner());
```